app/database/permission_granted_store_compute_all_access.go

```diff
@@ -47,63 +47,50 @@ func (s *PermissionGrantedStore) computeAllAccess() {
 	mustNotBeError(err)
 	defer func() { mustNotBeError(stmtDeleteProcessedChildren.Close()) }()
 
-	// computation for group-item pairs marked as 'self' in permissions_propagate (so normally all of them)
+	// computation for group-item pairs marked as 'self' in permissions_propagate (so all of them)
 	const queryUpdatePermissionsGenerated = `
 		INSERT INTO permissions_generated
 			(group_id, item_id, can_view_generated, can_grant_view_generated, can_watch_generated, can_edit_generated, is_owner_generated)
 		SELECT
 			permissions_propagate.group_id,
 			permissions_propagate.item_id,
-			IF(granted.is_owner, 'solution', GREATEST(
-				IFNULL(granted.can_view_value, 1),
-				IFNULL(new_data.can_view_value, 1)
+			IF(MAX(permissions_granted.is_owner), 'solution', GREATEST(
+				IFNULL(MAX(permissions_granted.can_view_value), 1),
+				IFNULL(MAX(
+					CASE
+					WHEN parent.can_view_generated IS NULL OR parent.can_view_generated IN ('none', 'info') THEN 1 /* none */
+					WHEN parent.can_view_generated = 'content' OR items_items.descendants_and_solution_view_propagation IS NULL OR
+					     items_items.descendants_and_solution_view_propagation = 'none' THEN
```

Comment:
`items_items.descendants_and_solution_view_propagation ` cannot be NULL if ` parent.can_view_generated` is not null (first WHEN), so I think this check should be removed (for readability as well)

Reply:
fixed